### PR TITLE
Upgrade Django

### PR DESCRIPTION
**Pull request by jabley**
I've broken this up into a sequence of commits.

- Upgrade to 1.7.0
  - Fix issues associated with that
- Upgrade to 1.7.8 as the most recent 1.7.x release

I've not upgraded to 1.8 since it still feels a little new.

I think the biggest functional change around this is the serialisation format
for Datasets. Tests were failing, since the serialised form was setting the
`bearer_token` field to `None` if it was an empty string. The model however
allows that. I wasn't sure which way to go with this inconsistency, so I've
changed the serialised form to be in-line with the model definition.

bc34629d describes previous problems with decisions in the model and serialisation.

### CODE REVIEW

**Comment by mattrco** [comment]
Tested locally and I'm happy this works as advertised.

It's unfortunate that we allow an empty string as the bearer token in the model. I'm on the fence about exposing it in serialized form though. What say ye, @timmow?

**Comment by jcbashdown** [comment]
Looking into this bearer token stuff...

File: README.md [hunk 1]
```diff
@@ -31,9 +31,9 @@ export DJANGO_SETTINGS_MODULE=stagecraft.settings.development
 
 Before the server is run for the first time the database needs to be synced.
 
-*Note: The first time you `syncdb`, you'll be asked to setup a `superadmin`.*
+*Note: The first time you `migrate`, you'll be asked to setup a `superadmin`.*
 ```
-python manage.py syncdb
+python manage.py migrate
```
Comment (jcbashdown) [inline README.md]: I am getting `Unknown command: 'migrate'`. syncdb seems to work though?

**Comment by jcbashdown** [comment]
Okay, found out a few things and there will be a pr in backdrop. I'm not 100% sure what your concern with serialising this was @mattrco but I'm basically happy to merge.

**Comment by mattrco** [comment]
:rocket:

**Pull request merged**